lightning/src/onion_message/messenger.rs

```diff
@@ -1145,7 +1140,14 @@ where
 				onion_routing_packet: outgoing_packet,
 			};
 
-			Ok(PeeledOnion::Forward(next_hop, onion_message))
+			let our_node_id = node_signer.get_node_id(Recipient::Node).unwrap();
+
+			match next_hop {
+				NextMessageHop::NodeId(ref id) if id == &our_node_id => {
+					peel_onion_message(&onion_message, secp_ctx, node_signer, logger, custom_handler)
```

Comment:
Is there any risk somehow of infinite recursion?

Reply:
I hadn't spotted a specific problem, was just curious how certain it is that infinite recursion cannot happen. If the explanation isn't straight-forward, I'd add it as a comment to the code. And refer to point where the limitation is, something with `new_packet_bytes` I think?

Reply:
It can't be infinite, but with an extra-long onion message ISTM it could still be a lot of processing time. If someone created a blinded path terminating at us with a bunch of dummy hops and we only find out that it's an inauthentic path when we get to the final layer, seems like that might be an issue. 

@shaavan would it be possible to explore adding a new `ControlTlvs::Dummy` variant that contains an HMAC and nonce, similar to how we authenticate blinded receive payloads elsewhere? And authenticate those fields before peeling the onion further?

Reply:
Sure! Let me give it a shot! 🚀

Reply:
Hello! I’ve updated the approach in [pr3728.04](https://github.com/shaavan/rust-lightning/commits/pr3728.04) to introduce `ControlTlvs::Dummy` with authentication.  
Thanks so much, @valentinewallace, for all your guidance offline!

Reply:
@jkczyz Wouldn't that have the same issue, if the first dummy hop is still too large and needs an extra padding block? Not too familiar with how padding works though.

Reply:
Yeah, @shaavan and I were just discussing that. The sender would gain that knowledge but not a forwarding node, IIUC. Maybe we could divide the authentication data across two hops to avoid this?

Reply:
Yeah, the problem is that dummy hops could be differentiated from forward hops (buy the sender) since the former would require two "blocks" of padding while the latter only one.